src/app/frontend/replicasetlist/logsmenu_controller.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Controller for the logs menu view.
+ *
+ * @final
+ */
+export default class LogsMenuController {
+  /**
+   * @param {!angular.$log} $log
+   * @param {!angular.$resource} $resource
+   * @ngInject
+   */
+  constructor($log, $resource) {
+    /** @private {!angular.$resource} */
+    this.resource_ = $resource;
+
+    /** @private {!angular.$log} */
+    this.log_ = $log;
+
+    /** @export {string} */
+    this.replicaSetName;
+
+    /** @export {string} */
+    this.namespace;
+
+    /** @export {!Array<!backendApi.ReplicaSetPodWithContainers>} */
+    this.replicaSetPodsList;
+  }
+
+  /**
+   * Opens menu with pods and link to logs.
+   * @param  {!function(MouseEvent)} $mdOpenMenu
+   * @param  {!MouseEvent} $event
+   * @export
+   */
+  openMenu($mdOpenMenu, $event) {
+    if (this.replicaSetPodsList) {
```

Comment:
Please add this explanation as `//` comment :) This would be useful for future people to know about.